src/pt/doramogo/src/eu/kanade/tachiyomi/animeextension/pt/doramogo/Doramogo.kt

```diff
@@ -0,0 +1,206 @@
+
+    override fun latestUpdatesNextPageSelector() = null
+
+    // =============================== Search ===============================
+    override suspend fun getSearchAnime(page: Int, query: String, filters: AnimeFilterList): AnimesPage {
+        return if (query.startsWith(PREFIX_SEARCH)) { // URL intent handler
+            val id = query.removePrefix(PREFIX_SEARCH)
+            client.newCall(GET("$baseUrl/dorama/$id"))
+                .awaitSuccess()
+                .use(::searchAnimeByIdParse)
+        } else {
+            super.getSearchAnime(page, query, filters)
+        }
+    }
+
+    private fun searchAnimeByIdParse(response: Response): AnimesPage {
+        val details = animeDetailsParse(response.asJsoup())
+        return AnimesPage(listOf(details), false)
+    }
+
+    override fun getFilterList() = DoramogoFilters.FILTER_LIST
+
+    override fun searchAnimeRequest(page: Int, query: String, filters: AnimeFilterList): Request {
+        val params = DoramogoFilters.getSearchParameters(filters)
+
+        val url = "$baseUrl/search".toHttpUrl().newBuilder()
+            .addPathSegment(query)
+            .addIfNotBlank("filter_audio", params.audio)
+            .addIfNotBlank("filter_genre", params.genre)
+            .build()
+
+        return GET(url, headers = headers)
+    }
+
+    override fun searchAnimeSelector() = popularAnimeSelector()
+
+    override fun searchAnimeFromElement(element: Element) = popularAnimeFromElement(element)
+
+    override fun searchAnimeNextPageSelector() = null
+
+    // =========================== Anime Details ============================
+    override fun animeDetailsParse(document: Document) = SAnime.create().apply {
+        setUrlWithoutDomain(document.location())
+        title = document.selectFirst("div.dados h1")!!.text()
+        thumbnail_url = document.select("div.image--cover").attr("style")
+            .substringAfter("background-image: url('").substringBefore("')")
+        description = document.selectFirst("p.readMor")!!.textNodes().joinToString("")
+    }
+
+    // ============================== Episodes ==============================
+    override fun episodeListParse(response: Response): List<SEpisode> {
+        return super.episodeListParse(response).reversed()
+    }
+
+    override fun episodeListSelector() = "li.episode--content"
+
+    override fun episodeFromElement(element: Element) = SEpisode.create().apply {
+        setUrlWithoutDomain(element.selectFirst("a")!!.attr("href"))
+        element.selectFirst("div.title-episode a")!!.textNodes().joinToString("").let {
+            name = it.substringAfter(". ")
+            episode_number = it.substringBefore(". ").toFloatOrNull() ?: 1F
+        }
+    }
+
+    // ============================ Video Links =============================
+    override fun videoListParse(response: Response): List<Video> {
+        val document = response.asJsoup()
+
+        val urls = document.select("div.source-box iframe").map {
```

Comment:
```suggestion
        val urls = document.select("div.source-box iframe[src]").map {
```
You should only select the iframes that actually has the src attribute, or else things might break